### DISCUSSION TITLE
#Livewire3 - How can I listen to events from a different component that is on a different page? like emitTo?

### DISCUSSION THREAD

**Discussion by fahrianggara** [livewire/livewire#8688] [1. Help]
Hello everyone, I have a question about **Livewire 3**. How can I listen to events from a different component that is on a different page? In other words, the components are not on the same page but on different routes.

For example, I'm dispatching from the `Home.php` component, and I want to send data from `Home.php` to the `Register.php` component (different pages). How can I make the dispatch from `Home.php` be listened to by the `Register.php` component?

I have followed the documentation here: [Livewire Events Documentation](https://livewire.laravel.com/docs/events#dispatching-directly-to-another-component), but the dispatch is still not being triggered. I've searched everywhere but haven't found a solution.

Has anyone used this kind of setup before? If so, could you share how you did it?

Thank you.

**Comment by Zeeshan607** [comment]
Hi @fahrianggara did you find the solution to this problem I am having same problem my both components are on different page and I want to dispatch event from one component to other which is on other page, but its not listening. if you found the solution to this I would appreciate the help
Thanks

**Reply by BVFI** [reply]
This issue lost me so much time, I couldnt believe that even after using dispatch()->to() the event didnt work. The components MUST be on the same page, which is really infuriating, beacause the developer should be able to manually handle this. The ->to() is the perfect function, yet it doesn't work, its basically useless. At this point I've given up figuring out a way to do this, Im sure theres a workaround, but I dont want code cluster, Id rather have duplicated code. Shame that this issue isnt more popular, eventually people will run into this issue, because theres nothing about it in the documentation. Duplicated code, here I come

**Reply by lrljoe** [reply]
Obviously the component needs to be on the same page... Not sure what isn't clear about that from the docs...  Perhaps you can point out where this isn't made clear?

When Livewire is dispatching to, it's all in the context of the current tab/window, as it is interacting with the DOM and JS.

Should you wish to dispatch across browser windows/tabs, then you need to use Echo or similar.

There are of course ways to load a new component, and pass the data in.

It seems like you're missing something basic if you're duplicating code

**Reply by BVFI** [reply]
The documentation doesnt really make it clear that it needs to be on the same page, its written very hastily and without clearance, or put in the wrong parts.
It never says same page, and it makes it sound very "robust" that can be listened "from any other component". It should be more specific and emphasized If its limited to the same page, should be the second thing written after explaining what an event is. Do you see how clearly you made it just by replying? Why isnt that important detail made more clear in the documentation? 
The code I have, in simple terms is, I have 2 components, one for indexing and one for creating/updating, however, I wish to call an event from the Index to the Create/Update (Components need to be in a different page). Now I have to copy the Create/Update function code and paste it in the Index, because they are on a different page. Sure, I can make a helper function, but the function code is very specific and I want my things to be as much centralized as possible.

**Reply by lrljoe** [reply]
For centralising your code, put it in a trait, and both components can use the same trait.  No copy/paste needed.

Now of course it depends on your design, but there's plenty of ways to have your "list", "create/edit" components on a single page, and swapping which component is visible.

Or you can move your "create/edit" into a Wire Elements Modal, which is my preferred approach, and just open the specific modal when the user wants to create/edit.  Or of course build your own modal component approach.
